app/static/src/app/components/sessions/SessionsPage.vue

```diff
@@ -101,6 +124,50 @@ export default defineComponent({
             toggleEditSessionLabelOpen(true);
         };
 
+        const ensureFriendlyId = async (session: SessionMetadata) => {
+            lastCopySessionId.value = session.id;
+            if (session.friendlyId) {
+                return session.friendlyId;
+            }
+            lastCopyMsg.value = "Fetching code...";
+            await store.dispatch(`sessions/${SessionsAction.GenerateFriendlyId}`, session.id);
+            await nextTick();
+            const friendlyId = sessionsMetadata.value.find((m: SessionMetadata) => m.id === session.id)?.friendlyId;
+            if (!friendlyId) {
+                lastCopyMsg.value = "Error fetching code";
+            }
```

Comment:
Instead of returning error message, shouldn't it default to unfriendly id ?

Reply:
No, we're not supporting sharting with 'unfriendly' ids (sessionIds) - this should only happen if there's network trouble.